Source/ProjectRimFactory/AutoMachineTool/Building_ItemPuller.cs

```diff
@@ -60,25 +77,21 @@ public override void ExposeData()
 
             base.ExposeData();
 
-            Scribe_Deep.Look<ThingFilter>(ref this.filter, "filter");
             Scribe_Values.Look<bool>(ref this.active, "active", false);
             Scribe_Values.Look<bool>(ref this.right, "right", false);
+            Scribe_Deep.Look(ref settings, "settings", this);
             Scribe_Values.Look<bool>(ref this.takeForbiddenItems, "takeForbidden", true);
-
-            if (this.filter == null) this.filter = new ThingFilter();
-
-            this.storageSettings = new StorageSettings { filter = this.filter };
         }
 
         public override void SpawnSetup(Map map, bool respawningAfterLoad)
         {
             base.SpawnSetup(map, respawningAfterLoad);
 
+            settings = GetStoreSettings();
```

Comment:
I am almost 100% sure this particular line isn't needed?  The first time something tries to GetStoreSettings() it will pop up...

Reply:
Oh, wait, I see why - we're accessing the `settings` field directly!

Reply:
So there's definitely a "code safety" value to using GetStoreSettings() internally as well, but it's (hopefully) not important enough to bother with, and this will be oh-sp-slightly faster.